packages/engine/Source/Core/Event.js

```diff
@@ -22,10 +22,10 @@ import defined from "./defined.js";
  * evt.removeEventListener(MyObject.prototype.myListener);
  */
 function Event() {
-  this._listeners = [];
-  this._scopes = [];
+  this._listeners = new Map(); // listener => Set of scopes
```

Comment:
Totally optional, but this could be formalized with a type definition:

```suggestion
  /**
   * @type {Map<Listener,Set<object>>}
   * @private
   */
  this._listeners = new Map();
```